nixos/tests/nixos-test-driver/tty.py

```diff
@@ -0,0 +1,26 @@
+import string
+import re
+
+test_string = "".join(c for c in string.printable if c not in string.whitespace)
+
+with subtest("wait_until_tty_matches"):
+    with subtest("success"):
+        machine.wait_until_tty_matches(1, "root@")
+
+    with subtest("failure"), no_sleep(), must_raise("action timed out"):
+        machine.wait_until_tty_matches(1, "asdfasdf", timeout=10)
+
+with subtest("send_chars"):
+    machine.send_chars("cat\n")
+    machine.sleep(2)
+    machine.send_chars(test_string + "\n")
+    machine.wait_until_tty_matches(1, re.escape(test_string), timeout=10)
+    machine.send_key("ctrl-d")
+
+with subtest("wait_for_console_text"):
+    with subtest("failure"), no_sleep(), must_raise("action timed out"):
+        machine.wait_for_console_text("foobar", timeout=10)
+
+#     with subtest("success"):
```

Comment:
Why is this commented out?

Reply:
Because wait_for_console_text did something else than I initially thought, and I haven't found a good way to test it yet.